src/handlers/Handlers.cpp

```diff
@@ -0,0 +1,159 @@
+#include <handlers/Handlers.h>
+#include <handlers/methods/Account.h>
+#include <handlers/methods/Channel.h>
+#include <handlers/methods/Exchange.h>
+#include <handlers/methods/Ledger.h>
+#include <handlers/methods/Subscribe.h>
+#include <handlers/methods/Transaction.h>
+#include <etl/ETLSource.h>
+
+namespace RPC
+{
+
+class HandlerTable
+{
+private:
+    HandlerTable()
+    {
+        for (auto v = RPC::ApiMinimumSupportedVersion;
+             v <= RPC::ApiMaximumSupportedVersion;
+             ++v)
+        {
+            addHandler<AccountChannels>(v);
+            addHandler<AccountCurrencies>(v);
+            addHandler<AccountInfo>(v);
+            addHandler<AccountLines>(v);
+            addHandler<AccountObjects>(v);
+            addHandler<AccountOffers>(v);
+            addHandler<AccountTx>(v);
+            addHandler<BookOffers>(v);
+            addHandler<ChannelAuthorize>(v);
+            addHandler<ChannelVerify>(v);
+            addHandler<Ledger>(v);
+            addHandler<LedgerData>(v);
+            addHandler<LedgerEntry>(v);
+            addHandler<LedgerRange>(v);
+            addHandler<Subscribe>(v);
+            addHandler<Unsubscribe>(v);
+            addHandler<Tx>(v);
+        }
+    }
+
+public:
+    static HandlerTable const&
+    instance()
+    {
+        static HandlerTable const handlerTable{};
+        return handlerTable;
+    }
+
+    Handler const*
+    getHandler(unsigned version, std::string name) const
+    {
+        if (version > RPC::ApiMaximumSupportedVersion ||
+            version < RPC::ApiMinimumSupportedVersion)
+            return nullptr;
+        auto& innerTable = table_[versionToIndex(version)];
+        auto i = innerTable.find(name);
+        return i == innerTable.end() ? nullptr : &i->second;
+    }
+
+private:
+    std::array<std::map<std::string, Handler>, APINumberVersionSupported>
+        table_;
+
+    template <class HandlerImpl>
+    void
+    addHandler(unsigned version)
+    {
+        assert(
+            version >= RPC::ApiMinimumSupportedVersion &&
+            version <= RPC::ApiMaximumSupportedVersion);
+        auto& innerTable = table_[versionToIndex(version)];
+        assert(innerTable.find(HandlerImpl::name()) == innerTable.end());
+
+        Handler h;
+        h.name = HandlerImpl::name();
+        h.role = HandlerImpl::role();
+
+        h.method = [](Context& ctx, boost::json::object& result) -> Status
+            {
+                HandlerImpl handler{ctx, result};
+
+                auto status = handler.check();
+                if (status)
+                    result = make_error(status.error);
+
+                return status;
+            };
+
+        innerTable[HandlerImpl::name()] = h;
+    }
+
+    inline unsigned
+    versionToIndex(unsigned version) const
+    {
+        return version - RPC::ApiMinimumSupportedVersion;
+    }
+};
```

Comment:
I know this code comes from rippled. I don't really like this code. I think a lot of it is overkill. I like what we had before, where we just have a map and a switch statement. Is there a reason we can't do that?

Reply:
No, we can do that, I just liked how rippled did it better. 

I was also considering adding RPC versioning in this PR. This also will be much more extendable to versioning, versus adding a `switch` for every new API version we want to add.